apps/bfd-db-migrator/src/main/java/gov/cms/bfd/migrator/app/ProcessOutputConsumer.java

```diff
@@ -0,0 +1,83 @@
+package gov.cms.bfd.migrator.app;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * Managing external processes is tricky: at the OS level, all processes' output is sent to a
+ * buffer. If that buffer fills up (because you're not reading the output), the process will block
+ * -- forever. To avoid that, it's best to always have a separate thread running that consumes a
+ * process' output. This {@link ProcessOutputConsumer} is designed to allow for just that.
+ */
+public final class ProcessOutputConsumer implements Runnable {
```

Comment:
is this a new class or a copy of existing code? If the latter, can we move to a shared location instead?

Reply:
Another copy; I'll add a TODO to move this to a common location as part of 1558 when I excise the old code from pipeline.